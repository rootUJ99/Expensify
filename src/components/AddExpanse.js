import React from 'react';
import ReactDOM from 'react-dom';
const AddExpanse = () => (
    <div>
        this is add page
    </div>
);
export default AddExpanse;