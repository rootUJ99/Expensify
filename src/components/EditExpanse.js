import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
export class EditExpanse extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
        console.log('updated', expense);
    }
    onRemove = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id })
        this.props.history.push('/');
    }
    render() {
        return (
            <div>
                <div className="page-header" >
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
                <button
                    className="button2 button2--remove"
                    onClick={this.onRemove}
                > Remove Expense</button>
            </div>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    };
};
const mapDispathToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});
export default connect(mapStateToProps, mapDispathToProps)(EditExpanse);