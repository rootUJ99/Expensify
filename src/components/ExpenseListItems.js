import React from 'react';
import {Link} from 'react-router-dom';
const ExpenseListItem = ({ id, dispatch, description, amount, createdAt})=>(
    <div>
    <h3><Link to={`/edit/${id}`}>{description}</Link></h3>
    <h4>{amount}-{createdAt}</h4>
       
    </div>
);

export default ExpenseListItem;
