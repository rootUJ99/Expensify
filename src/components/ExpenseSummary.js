import React from 'react';
import ReactDOM from 'react-dom';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
export const ExpenseSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal).format('$0,0.00');
    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
        </div>
    );
}
const mapStateToProps=(state)=>{
    const visiblityExpenses = selectExpenses(state.expenses,state.filters);
    return{
        expenseCount:visiblityExpenses.length,
        expensesTotal:selectExpensesTotal(visiblityExpenses)
    };
};
export default connect(mapStateToProps)(ExpenseSummary);
