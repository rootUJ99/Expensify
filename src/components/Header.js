import React from 'react';
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth';
export const Header = ({startLogout}) => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/dashboard" activeClassName="active-nav">Dashboard</NavLink>
        <NavLink to="/create" activeClassName="active-nav">Create Expense</NavLink>
        <button onClick={startLogout}>Logout</button>
    </header>
);
const mapDispatchProps=(dispatch)=>({
    startLogout:()=>dispatch(startLogout())
})
export default connect(undefined, mapDispatchProps)(Header);