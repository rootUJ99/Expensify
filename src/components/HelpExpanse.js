import React from 'react';
import ReactDOM from 'react-dom';
const HelpExpanse = () => (
    <div>
        this is help
    </div>
);
export default HelpExpanse;