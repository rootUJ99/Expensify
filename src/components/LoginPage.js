import React, { Component} from 'react';
import { connect } from 'react-redux';
import {startLogin} from '../actions/auth';

export const LoginPage=({startLogin})=> (
            <div>
                <button onClick={startLogin}>Login</button>
            </div>
        )
const matchDispatchProps=(dispatch)=>({
    startLogin:()=>dispatch(startLogin())
});
export default connect(undefined,matchDispatchProps)(LoginPage);