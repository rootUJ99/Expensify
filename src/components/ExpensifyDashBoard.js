import React from 'react';
import ReactDOM from 'react-dom';
import ExpenseList from './ExpenseList';
import ExpenseListItems from './ExpenseListItems';
import ExpenseListFilters from './ExpenseListFilters';
const ExpensifyDashBoard = () => (
    <div>
        <ExpenseListFilters/>
        <ExpenseList/>
    </div>
);
export default ExpensifyDashBoard;