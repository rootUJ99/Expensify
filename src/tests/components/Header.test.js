import React from 'react';
import {Header} from '../../components/Header';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';
//test header component
test('test for header',()=>{
    const wrapper = shallow(<Header startLogout={()=>{}}/>);
    expect(wrapper).toMatchSnapshot();
    //expect(wrapper.find('h1').text()).toBe('Expensify');
});

//should log out on button click
test('should log out on button click',()=>{
    const startLogout=jest.fn();
    const wrapper = shallow(<Header startLogout={startLogout}/>);
    wrapper.find('button').simulate('click');
    expect(startLogout).toHaveBeenCalled();
});