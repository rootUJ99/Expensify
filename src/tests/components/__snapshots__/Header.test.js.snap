// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for header 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="active-nav"
    ariaCurrent="true"
    to="/dashboard"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="active-nav"
    ariaCurrent="true"
    to="/create"
  >
    Create Expense
  </NavLink>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
